src/Data/AudienceFolder.elm

```diff
@@ -23,6 +30,44 @@ type alias AudienceFolder =
     }
 
 
+-- = BASIC FIELDS DECODERS =
+idField = D.field "id" D.int
+nameField = D.field "name" D.string
+parentField = D.field "parent" (D.nullable D.int)
+
+-- = AUDIENCE FOLDER DECODER --
+folderDecoder : Decoder AudienceFolder
+folderDecoder =
+  D.succeed AudienceFolder
+    |> D.andMap idField
+    |> D.andMap nameField
+    |> D.andMap parentField
+
+foldersDecoder : Decoder (List AudienceFolder)
+foldersDecoder = D.field "data" (D.list folderDecoder)
+
+maudienceFolders0 : Result D.Error (List AudienceFolder)
+maudienceFolders0 = D.decodeString foldersDecoder audienceFoldersJSON
```

Comment:
I'm unsure of what the `0` suffix naming means. I believe I haven't seen that even in Haskell codebases etc. Can you elaborate?

Reply:
That's just a personal convention. I usually give a suffix (in math notation it would actually be subscript) to variable names that denote "concrete" (or somehow static, persistant) elements in the current context, e.g. I would write `n0 = 123`, or `verySpecificString1 = "hello, world"`. I use this most often in unit tests. But the parameters of some function `f : Int -> a` wouldn't have such a suffix `f n = ...`. The distinction of "concrete" vs "general" variable names is not formal, it's more what feels right to me. In the real codebase we wouldn't have a static json file, we would get it via HTTP request, so the module `Data/AudienceFolder ` would export some sort of a function and not a concrete lists. So I wouldn't actually have to use something like this.

Also it seems that Elm doesn't allow variable shadowing, so that gave me a bunch of headaches - because of this I was forced to use something like suffixes. But later I managed to refactor the code such that it didn't use shadowing (so maybe the ban is not completely evil :smile:  )

Reply:
Yeah the shadowing ban is a bit annoying but it has helped us find some bugs too! So in the end I think I like it :smile: